internal/sysfs/osfile.go

```diff
@@ -275,6 +275,9 @@ func (f *osFile) Pwrite(buf []byte, off int64) (n int, errno experimentalsys.Err
 
 // Truncate implements the same method as documented on sys.File
 func (f *osFile) Truncate(size int64) (errno experimentalsys.Errno) {
+	if size < 0 {
```

Comment:
Only Go 1.24 windows has a different result here. But I couldn't see a reason not to do parameter validation directly here